metricflow/test/fixtures/model_fixtures.py

```diff
@@ -42,6 +49,23 @@ def _data_set_to_source_nodes(
     return source_node_builder.create_from_data_sets(list(data_sets.values()))
 
 
+def query_parser_from_yaml(
+    yaml_contents: List[YamlConfigFile], time_spine_source: TimeSpineSource
+) -> MetricFlowQueryParser:
+    """Given yaml files, return a query parser using default source nodes, resolvers and time_sp"""
+    semantic_model = SemanticModel(parse_yaml_files_to_validation_ready_model(yaml_contents).model)
```

Comment:
You might want to run the semantic validators on this, that way people get better error messages when they write a test model that's invalid.

You can either refactor dir_to_model a tiny bit more or add a call to `ModelValidator().checked_validations()` here.